exercises/palindrome-products/PalindromeProductsExample.swift

```diff
@@ -31,7 +27,6 @@ struct PalindromeProducts {
     private func calculate(_ upTo: Mode) -> Palindrome {
 
         let rangeOuter = minFactor...maxFactor
-        var multiplications = [Palindrome]()
 
         //Multithreaded code
         var results = [[Palindrome]](repeating: [Palindrome](), count: rangeOuter.count)
```

Comment:
I believe the issue with the failing test has to do with the ``results`` array. 
http://stackoverflow.com/questions/28191079/create-thread-safe-array-in-swift/28976644#28976644
If the array were to be written to from two threads at the same time, it will have problems and this is why it doesn't happen often. 
https://www.reddit.com/r/swift/comments/4myh3w/is_swift_array_thread_safe_can_swift_array_face/

we need to lock the array when writing to it to prevent other threads from writing to it at the same time. 
https://www.bignerdranch.com/blog/locking-in-swift-helping-protect-me-from-myself/